x-pack/plugins/security_solution/public/resolver/view/font_size.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * Return a font-size based on a scale, minimum size, and a coefficient.
+ * @deprecated The projection matrix is already scaled non-linearly.
```

Comment:
@michaelolo24 @bkimmel I've been thinking about this.

The zoom feature in resolver defines non-linear scale. It affects the position of nodes. It could make sense that the size of graphical elements as well as the size of text doesn't scale equally. However I think we should define this in one place, like we do with the zoom scale.

If we do that, we can define the size of stuff like this:

```
const worldUnitInPixelsAtMinimumZoom = 100
const baseFontSizeAtMinimumZoom = 10

const worldUnitInPixelAtMaximumZoom = 500
const baseFontSizeAtMaximumZoom = 24

const geometricCommonRatio = 2.34 // made up number that describes the curve.
```

The `baseFontSize...` variables would also control the size of things like borders, padding, and other graphical units.


This approach should be easy to communicate with design.

Reply:
So you would have some kind of curve function you'd plug like min, max, commonRatio and current xScale... something like
`const fontSize = fromCurve(commonRatio, min, max, currentScale)` ?

Reply:
Or excuse me if I"m being mathematically illiterate there...

Reply:
yea thats what i mean. basically we'd set a minimum and max (for the map itself and for graphical stuff like font and icons) and then they'd scale the same (non-linear) way.

Reply:
That makes sense. It would make it clearer to identify any relationships between scaling of the fonts and graphical elements. This will make it easier to clean up the magic number logic in the `process_event_dot` file as well